### PR TITLE
Upd/`neo-go` subscriptions

### PR DESCRIPTION
**Pull request by carpawell**
Only adopts new API. More adds that removes; could look strange a little. But suggest separating updates and refactors.

### CODE REVIEW

File: pkg/morph/client/multi.go [hunk 1]
```diff
@@ -156,11 +185,12 @@ mainLoop:
 			return
 		case <-t.C:
 			c.switchLock.RLock()
+
```
Comment (fyrchik) [inline pkg/morph/client/multi.go:188]: ?

File: pkg/morph/client/multi.go [hunk 2]
```diff
@@ -156,11 +185,12 @@ mainLoop:
...
 			endpointsCopy := make([]Endpoint, len(c.endpoints.list))
 			copy(endpointsCopy, c.endpoints.list)
-
```
Comment (fyrchik) [inline pkg/morph/client/multi.go:161]: ??

File: pkg/morph/client/notifications.go [hunk 3]
```diff
@@ -214,72 +234,109 @@ func (c *Client) restoreSubscriptions(cli *rpcclient.WSClient, endpoint string)
...
+				// background client (test) switch, no need to send
+				// any notification, just preventing dead-lock
+				continue
 			}
+
+			c.routeEvent(e)
 		}
 	}()
 
+	if background {
```
Comment (fyrchik) [inline pkg/morph/client/notifications.go:273]: If it is false, do can we take a mutex too?
It could look more readable without `if`.

**Review by fyrchik — COMMENTED** [review]
Correct me if I am wrong, in this PR our main goal is to remove some of the `deprecated` messages?

**Review by carpawell — COMMENTED** [review]
> Correct me if I am wrong, in this PR our main goal is to remove some of the deprecated messages?

@fyrchik, yes, all the "deprecated" msgs about the notification processes are fixed (but that is not a refactor and that is not a PR that fixes _all_ the linter messages)

File: pkg/morph/client/notifications.go (hunk 3 above)
Comment (carpawell) [reply pkg/morph/client/notifications.go:273]: no, we would have a dead-lock then: `background` flag is `false` only if we have already lost connection and try to establish a new one, so `switchLock` has been already taken

File: pkg/morph/client/multi.go (hunk 1 above)
Comment (carpawell) [reply pkg/morph/client/multi.go:188]: could not find unlocking line while was preparing that PR so decide to chose a better (IMO) spacing

Comment (carpawell) [reply pkg/morph/client/multi.go:188]: a separate commit/keep as it was?

Comment (fyrchik) [reply pkg/morph/client/multi.go:188]: I would keep if it then linter doesn't complain.

File: pkg/morph/client/client.go [hunk 4]
```diff
@@ -69,17 +72,20 @@ type Client struct {
...
+	// channels for ws notifications; protected with switchLock
+	notifications   chan rpcclient.Notification
+	blockRcv        chan *block.Block
+	notificationRcv chan *state.ContainedNotificationEvent
+	notaryReqRcv    chan *result.NotaryRequestEvent
 
 	// cached subscription information
 	subscribedEvents       map[util.Uint160]string
 	subscribedNotaryEvents map[util.Uint160]string
 	subscribedToNewBlocks  bool
```
Comment (fyrchik) [inline pkg/morph/client/client.go:84]: Can we embed `subsInfo` here? It allows us to get rid of `updateSubs` and possibly prevents us from forgetting something to copy when we add it.

**Review by fyrchik — APPROVED** [review]

**Review by acid-ant — APPROVED** [review]

**Pull request closed without merging**